core/che-core-api-core/src/main/java/org/eclipse/che/api/core/util/Version.java

```diff
@@ -0,0 +1,246 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.core.util;
+
+import javax.validation.constraints.NotNull;
+import java.util.Comparator;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.lang.Integer.parseInt;
+import static java.lang.String.format;
+import static java.util.regex.Pattern.compile;
+
+/**
+ * Version format:
+ * {MajorVersion}.{MinorVersion}.{BugFixVersion}.{HotFixVersion}-M{MilestoneVersion}-beta-{BetaVersion}-RC-SNAPSHOT.
+ * </p>
+ * Mandatory parts are only major, minor and bug fix versions.
+ *
+ * @see #toString()
+ * @see #VERSION
+ *
+ * @author Anatoliy Bazko
+ */
+public class Version implements Comparable<Version> {
+
+    private static final String MILESTONE_VERSION_SUFFIX = "-M";
+    private static final String BETA_VERSION_SUFFIX      = "-beta-";
+
+    private static final Pattern VERSION =
+            compile("^(0|[1-9]+[0-9]*)\\.(0|[1-9]+[0-9]*)\\.(0|[1-9]+[0-9]*)(\\.0|\\.[1-9]+[0-9]*|)" +
+                    "(" + MILESTONE_VERSION_SUFFIX + "[1-9]+[0-9]*|)" +
+                    "(" + BETA_VERSION_SUFFIX + "[1-9]+[0-9]*|)" +
+                    "(-RC|)" +
```

Comment:
There should not be any difference but for me `(-RC)?` looks better than `(-RC|)` here